var/spack/repos/builtin/packages/octave/package.py

```diff
@@ -182,12 +184,19 @@ def configure_args(self):
                 "--with-java-includedir=%s" % spec['jdk'].prefix.include,
                 "--with-java-libdir=%s"     % spec['jdk'].prefix.lib
             ])
+        else:
+            config_args.append("--disable-java")
 
         if '~opengl' in spec:
             config_args.extend([
                 "--without-opengl",
                 "--without-framework-opengl"
             ])
+        else:
+            config_args.extend([
+                "--without-opengl",
+                "--without-framework-opengl"
```

Comment:
judging from other inputs i think one needs to provide directories, i.e.
```
                "--with-glpk-includedir=%s" % spec['glpk'].prefix.include,
                "--with-glpk-libdir=%s"     % spec['glpk'].prefix.lib
```
but i don't see `opengl` package in Spack.